.github/workflows/release.yml

```diff
@@ -0,0 +1,42 @@
+name: Semantic Release
+on:
+  push:
+    branches:
+      - main
```

Comment:
it will run on each commit, how do you prevent the infinite loop of new versions?

Reply:
Should be ok

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run.

https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow#triggering-a-workflow-from-a-workflow